core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -605,6 +606,18 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  def setExec(dest: File): Unit = {
+    val files = dest.listFiles()
+    for (i <- files) {
+      if (i.isDirectory) {
+        setExec(i)
+      }
+      else {
+        i.setExecutable(true)
```

Comment:
No.. I think we shouldn't just change file permissions to be all executable blindly. Some of them might be intentionally not executable to prevent users mistakenly run. Is there any way to keep and restore the original file permissions?

Reply:
Not to mention, it seems like a huge security risk.

Reply:
+1.

Reply:
I believe there is consensus here that, in the current form, it is no go. As far as I can tell Hadoop `FileUtil` doesn't seem to handle info-zip extensions, neither does `java.util.zip`.  I am somewhat skeptical about bringing new dependencies just for the sake of this,especially, when better alternatives for distributing Python environments are already supported (pex, conda pack, and so on).

Python standard library [provides nice implementation](https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/zipfile.py#L545) of metadata extraction, that supports, among other things, file permissions, but implementing it on our side could be an overkill.

If the problem is actually `"spark.pyspark.python` then we might consider changing permissions only for this single path, but not before we have more eyes on this, to be sure it won't backfire.